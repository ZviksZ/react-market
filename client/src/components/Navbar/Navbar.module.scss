.navbar {
    .navbarToolbar {
        flex-wrap: wrap;
        background: #7b7b7b;
    }
    .navbarTitle {
        flex-grow: 1;
        color: #ffffff;
        &:hover {
            text-decoration: none;
        }
    }
    .navbarLink {
        margin: 0 15px;
        color: #ffffff;
    }
}
