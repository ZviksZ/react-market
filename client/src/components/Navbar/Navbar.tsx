import * as React from 'react'
import { AppBar } from '@material-ui/core'
import Toolbar from '@material-ui/core/Toolbar'
import Button from '@material-ui/core/Button'
import s from './Navbar.module.scss'
import { useState } from 'react'
import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart'
import Badge from '@material-ui/core/Badge'
import { NavLink } from 'react-router-dom'
import { useSelector } from 'react-redux'
import { NavbarCart } from './NavbarCart/NavbarCart'
import { selectCart, selectCartProductsSum } from '../../store/ducks/cart/selectors'

interface Props {
	setLogin: (param: boolean) => void
	setRegister: (param: boolean) => void
}

export const Navbar: React.FC<Props> = ({ setLogin, setRegister }) => {
	const [basketOpen, setBasketOpen] = useState<boolean>(false)
	const { cart } = useSelector(selectCart)
	const cartSum = useSelector(selectCartProductsSum)
	const cartCount = cart.reduce((acc, item) => acc + item.count, 0)

	return (
		<>
			<AppBar position="fixed" className={s.navbar}>
				<Toolbar className={s.navbarToolbar}>
					<NavLink to={'/'} className={s.navbarTitle}>
						React e-commerce
					</NavLink>
					<nav>
						<NavLink className={s.navbarLink} to={'/catalog'}>
							Catalog
						</NavLink>
					</nav>
					<Button onClick={() => setBasketOpen((prev) => !prev)} data-testid="open-basket" variant="contained" color="primary" className={s.navbarBasketBtn}>
						<Badge badgeContent={cartCount} color="secondary">
							<AddShoppingCartIcon fontSize={'small'} />
						</Badge>
					</Button>
					<div className={s.navabarLR}>
						<div className={s.navbarSmallLink} onClick={() => setLogin(true)}>
							Login
						</div>
						/
						<div className={s.navbarSmallLink} onClick={() => setRegister(true)}>
							Register
						</div>
					</div>

					{basketOpen && <NavbarCart cartSum={cartSum} setOpen={setBasketOpen} cartItems={cart} cartCount={cartCount} />}
				</Toolbar>
			</AppBar>
		</>
	)
}
