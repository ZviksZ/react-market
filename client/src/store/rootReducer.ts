import { combineReducers } from 'redux'
import { cartReducer } from './ducks/cart/reducer'
import { productsReducer } from './ducks/product/reducer'
import { authReducer } from './ducks/auth/reducer'

export const rootReducer = combineReducers({
	products: productsReducer,
	cart: cartReducer,
	auth: authReducer,
})
