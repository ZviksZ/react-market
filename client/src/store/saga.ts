import { all } from 'redux-saga/effects'
import { productsSaga } from './ducks/product/sagas'

export default function* rootSaga() {
	yield all([productsSaga()])
}
