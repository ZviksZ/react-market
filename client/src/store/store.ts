import { applyMiddleware, compose, createStore } from 'redux'
import createSagaMiddleware from 'redux-saga'
import { rootReducer } from './rootReducer'
import rootSaga from './saga'
import { ProductsState } from './ducks/product/contracts/state'
import { CartState } from './ducks/cart/contracts/state'
import { AuthState } from './ducks/auth/contracts/state'

const composeEnhancers = (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose

const sagaMiddleware = createSagaMiddleware()

export interface RootState {
	products: ProductsState
	cart: CartState
	auth: AuthState
}

export const store = createStore(rootReducer, composeEnhancers(applyMiddleware(sagaMiddleware)))

sagaMiddleware.run(rootSaga)
