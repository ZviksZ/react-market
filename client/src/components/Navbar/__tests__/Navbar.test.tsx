// @ts-nocheck
import * as React from 'react'
import { Navbar } from '../Navbar'
import { shallow } from 'enzyme'
import { store } from '../../../store/store'
import { Provider } from 'react-redux'

describe('testing Navbar component', () => {
	it('should button click change state', () => {
		const navbar = shallow(
			<Provider store={store}>
				<Navbar />
			</Provider>,
		)

		expect(navbar).toMatchSnapshot()

		/*const stateSetter = jest.fn()
		const handleClick = jest.spyOn(React, 'useState')
		handleClick.mockImplementation((state) => [state, stateSetter])

		const btn = getByTestId('open-basket')
		fireEvent.click(btn)

		btn.simulate('click')

		expect(stateSetter).toBeTruthy()*/
	})
})
