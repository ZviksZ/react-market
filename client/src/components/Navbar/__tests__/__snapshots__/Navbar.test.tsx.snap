// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Navbar component should button click change state 1`] = `ShallowWrapper {}`;
